### PR TITLE
test: add usersets tuple_cycle2 and tuple_cycle3 tests

### PR DESCRIPTION
**Pull request by ewanharris**
## Description

Adds tests that cover the `tuple_cycle2` and `tuple_cycle3` relations in usersets, verified the cycle cases by logging in the cycle detection logic.

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: tests/check/check_userset.go [hunk 1]
```diff
@@ -997,4 +997,73 @@ var usersetCompleteTestingModelTest = []*stage{
...
 		},
 	},
+	{
+		Name: "usersets_tuple_cycle2",
+		Tuples: []*openfgav1.TupleKey{
+			// user exists so no cycle
+			{Object: "directs-user:utc2_1", Relation: "tuple_cycle2", User: "user:utc2_1"},
+			{Object: "ttus:utc2_1", Relation: "direct_parent", User: "directs-user:utc2_1"},
+			{Object: "usersets-user:utc2_1", Relation: "tuple_cycle2", User: "ttus:utc2_1#tuple_cycle2"},
+			{Object: "directs-user:utc2_2", Relation: "tuple_cycle2", User: "usersets-user:utc2_1#tuple_cycle2"},
```
Comment (adriantam) [inline tests/check/check_userset.go:1007]: I think to actually make it a full cycle (but with exit), you need 
```
{Object: "directs-user:utc2_1", Relation: "tuple_cycle2", User: "usersets-user:utc2_1#tuple_cycle2"},
```

See https://github.com/openfga/openfga/blob/c2ef51f0c3fd80a62bd7ceed29360de5984b73d5/tests/check/complexity_three.go#L758-L763 as example

File: tests/check/check_userset.go [hunk 2]
```diff
@@ -997,4 +997,73 @@ var usersetCompleteTestingModelTest = []*stage{
...
+	},
+	{
+		Name: "usersets_tuple_cycle3",
+		Tuples: []*openfgav1.TupleKey{
+			// user exists so no cycle
+			{Object: "ttus:utc3_1", Relation: "userset_parent", User: "usersets-user:utc3_1"},
+			{Object: "complexity3:utc3_1", Relation: "cycle_nested", User: "ttus:utc3_1#tuple_cycle3"},
+			{Object: "directs-user:utc3_1", Relation: "tuple_cycle3", User: "user:utc3_1"},
+			{Object: "usersets-user:utc3_1", Relation: "tuple_cycle3", User: "directs-user:utc3_1#compute_tuple_cycle3"},
+			{Object: "directs-user:utc3_2", Relation: "tuple_cycle3", User: "complexity3:utc3_1#cycle_nested"},
```
Comment (adriantam) [inline tests/check/check_userset.go:1042]: Similar to above, to make it a cycle but with exit, you will want 
{Object: "directs-user:utc3_1", Relation: "tuple_cycle3", User: "complexity3:utc3_1#cycle_nested"},

File: tests/check/check_userset.go [hunk 3]
```diff
@@ -997,4 +997,72 @@ var usersetCompleteTestingModelTest = []*stage{
...
 		},
 	},
+	{
+		Name: "usersets_tuple_cycle2",
+		Tuples: []*openfgav1.TupleKey{
+			// user exists so no cycle
+			{Object: "directs-user:utc2_1", Relation: "tuple_cycle2", User: "user:utc2_1"},
+			{Object: "ttus:utc2_1", Relation: "direct_parent", User: "directs-user:utc2_1"},
+			{Object: "usersets-user:utc2_1", Relation: "tuple_cycle2", User: "ttus:utc2_1#tuple_cycle2"},
+			{Object: "directs-user:utc2_2", Relation: "tuple_cycle2", User: "usersets-user:utc2_1#tuple_cycle2"},
```
Comment (adriantam) [inline tests/check/check_userset.go:1007]: Do we still need directs-user:utc2_2?

File: tests/check/check_userset.go [hunk 4]
```diff
@@ -997,4 +997,72 @@ var usersetCompleteTestingModelTest = []*stage{
...
+				Expectation: true,
+			},
+			{
+				Name:        "cycle",
+				Tuple:       &openfgav1.TupleKey{Object: "usersets-user:utc3_4", Relation: "tuple_cycle3", User: "user:utc3_1"},
+				Expectation: false,
+			},
+			{
+				Name:        "invalid_user",
+				Tuple:       &openfgav1.TupleKey{Object: "usersets-user:utc3_3", Relation: "tuple_cycle3", User: "user:utc3_2"},
```
Comment (miparnisari) [inline tests/check/check_userset.go:1063]: There is no tuple mentioning `Object: "usersets-user:utc3_3"`. is this on purpose?

File: tests/check/check_userset.go (hunk 3 above)
Comment (miparnisari) [reply tests/check/check_userset.go:1007]: I think not, this and the tuple below are not used

**Review by adriantam — APPROVED** [review]

**Pull request merged**